wallet/wallet.c

```diff
@@ -595,6 +595,42 @@ wallet_htlc_sigs_load(const tal_t *ctx, struct wallet *w, u64 channelid)
 	return htlc_sigs;
 }
 
+bool wallet_remote_ann_sigs_load(struct wallet *w, u64 id,
+				 secp256k1_ecdsa_signature **remote_ann_node_sig,
+				 secp256k1_ecdsa_signature **remote_ann_bitcoin_sig)
+{
+	sqlite3_stmt *stmt;
+	int res;
+	stmt = db_select_prepare(w->db,
+				 "remote_ann_node_sig, remote_ann_bitcoin_sig"
+				 " FROM channels WHERE id = ?");
+	sqlite3_bind_int64(stmt, 1, id);
+
+	res = sqlite3_step(stmt);
+
+	/* This must succeed, since we know the channel exists */
+	assert(res == SQLITE_ROW);
+
+	if (sqlite3_column_type(stmt, 0) != SQLITE_NULL) {
+		if (!sqlite3_column_signature(stmt, 0, *remote_ann_node_sig)) {
+			db_stmt_done(stmt);
```

Comment:
In other parts of the code we use a `fail:` label to clear and return if we have multiple similar failure paths in a single function.

Reply:
Yes, now I use `fail:`